### PR TITLE
Oxi update

### PR DESCRIPTION
**Pull request by nikokaoja**
- We are now using RDF datasets (namedgraphs) instead of simple graphs, enabling support for multi space
  resolution of nodes and edges in CDF.
- Fixed bug in inference of multi-value types
- Bump dep on pyoxi to 0.4.3 and oxrdflib to 0.4.0
- Increase speed in loading RDF files in oxistore

### CODE REVIEW

File: cognite/neat/_rules/importers/_rdf/_inference2rules.py [hunk 1]
```diff
@@ -212,32 +213,12 @@ def _to_rules_components(
                     ):
                         properties[id_]["max_count"] = max(properties[id_]["max_count"], definition["max_count"])
 
-        # Add comments
```
Comment (nikokaoja) [inline cognite/neat/_rules/importers/_rdf/_inference2rules.py:215]: not used at all

File: cognite/neat/_graph/transformers/_prune_graph.py [hunk 2]
```diff
@@ -123,4 +123,4 @@ def transform(self, graph: Graph) -> None:
             for node in nodes_without_neighbours:
                 # Remove node and its property triples in the graph
                 if isinstance(node, ResultRow):
-                    graph.remove(triple=(node["subject"], None, None))
```
Comment (nikokaoja) [inline cognite/neat/_graph/transformers/_prune_graph.py:126]: new rdflib does not like this

File: cognite/neat/_store/_base.py [hunk 3]
```diff
@@ -244,33 +244,36 @@ def read(self, class_: str) -> Iterable[tuple[str, dict[str | InstanceType, list
...
+            By default we are using non-transactional mode for parsing RDF files.
+            This gives us a significant performance boost when importing large RDF files.
+            Underhood of rdflib we are triggering oxrdflib plugin which in respect
+            calls `bulk_load` method from oxigraph store. See more at:
+            https://pyoxigraph.readthedocs.io/en/stable/store.html#pyoxigraph.Store.bulk_load
         """
 
         # Oxigraph store, do not want to type hint this as it is an optional dependency
         if type(self.graph.store).__name__ == "OxigraphStore":
-
```
Comment (nikokaoja) [inline cognite/neat/_store/_base.py:260]: low-level code of oxistore is not anymore needed after update to latest oxrdflib and pyoxi

File: tests/tests_unit/rules/test_models/test_data_types.py [hunk 4]
```diff
@@ -75,48 +69,40 @@ def test_with_without_unit_not_equal(self) -> None:
                 {
                     "property": "a_boolean",
                     "valueType": "boolean",
-                    "Reference": "power:GeneratingUnit(property=activePower)",
```
Comment (nikokaoja) [inline tests/tests_unit/rules/test_models/test_data_types.py:78]: not needed anymore

**Review by doctrino — APPROVED** [review]
Even more code gone :D

**Pull request merged**